source/particle/generator/probability_density_function.cc

```diff
@@ -68,11 +68,13 @@ namespace aspect
         // the weights of all processes with a lower rank
         double local_start_weight = 0.0;
         MPI_Scan(&local_weight_integral, &local_start_weight, 1, MPI_DOUBLE, MPI_SUM, this->get_mpi_communicator());
```

Comment:
Just as a follow-up: In deal.II, this call has been replaced by `MPI_Exscan`. If you felt ambitious, feel free to change it here as well :-)

Reply:
Yes, Rene and I discussed that a bit after I made the pull request, but I ended up not changing it. I could make a new pull request for that, should not be that much work to bring it in line with deal.ii.

Reply:
done #3353 :)